projects/packages/waf/actions.php

```diff
@@ -20,6 +28,9 @@
 add_action(
 	'plugin_loaded',
 	function () {
+		if ( Waf_Runner::has_version_changed() ) {
```

Comment:
Yeah, I'm not sure we want this check to happen here. As you say, it'll only happen when our version constant changes, but that's still a check we don't need for 99% of requests. There's `https://developer.wordpress.org/reference/hooks/upgrader_process_complete/` although note the caveat there that if you're checking for your own plugin, the code that's running is the _old_ code.

Reply:
I've looked into that, but the behavior is not what we're actually after. We only wanted to generate/update the rules.php file when our rules have changed. The `upgrader_process_complete` hook would regenerate the the file whenever Jetpack gets updated. Checking the version on every request is a fairly common approach to solving this problem as it's just a single query.

Alternatively, I suppose we could combine `upgrader_process_complete` and the current approach where we only check the version if the plugin has been updated (as that would be the only time the hardcoded version would change).

However, considering that we are planning to implement a way to update rules whenever we want ([Asana Task](https://app.asana.com/0/1201069996155217/1201965730421001)), the new approach will deprecate the current one anyhow.

Reply:
> We only wanted to generate/update the rules.php file when our rules have changed.

Right, while the approach in this PR will update them when the hardcoded version in the plugin changes. This is potentially useful to solve a different problem (generated rules file trying to call old code in the plugin), although we'll have to be defensive about changes of that nature anyway. If that's what we want, I'd probably check in a shutdown hook instead to avoid holding up the current request. Or if this code will be deprecated shortly anyway, we could remove it?

Reply:
I've combined the `upgrader_process_complete` hook with our version number for now. It seems best to include some ability to updating the rules for now even if it will be deprecated soon. But by combining the two, at least the check won't happen on every load.